src/main/java/org/candlepin/subscriptions/metering/service/prometheus/promql/QueryBuilder.java

```diff
@@ -61,25 +78,64 @@ public String buildAccountLookupQuery(QueryDescriptor queryDescriptor) {
       throw new IllegalArgumentException(
           String.format("Unable to find account query template for key: %s", templateKey));
     }
-    log.debug("Building account lookup PromQL.");
+    LOGGER.debug("Building account lookup PromQL.");
     return buildQuery(template.get(), queryDescriptor);
   }
 
-  private String buildQuery(String template, QueryDescriptor descriptor) {
-    ExpressionParser parser = new SpelExpressionParser();
-    StandardEvaluationContext context = new StandardEvaluationContext(descriptor);
-
-    // Only allow nested expressions based on a config setting. We need to do this
-    // to prevent potential infinite recursion.
-    String query = template;
-    for (int i = 0; i < metricProperties.getTemplateParameterDepth(); i++) {
-      query = (String) parser.parseExpression(query, new TemplateParserContext()).getValue(context);
-      if (query == null) {
-        throw new IllegalStateException(
-            String.format("Unable to parse query template! %s", template));
+  private String buildQuery(String query, QueryDescriptor descriptor) {
+    // Support of system properties binding in the format of "${KEY:default value}"
+    if (query.contains(SYSTEM_PROPERTY_OPEN_TAG)) {
+      String rawKey =
+          StringUtils.substringBetween(query, SYSTEM_PROPERTY_OPEN_TAG, SYSTEM_PROPERTY_CLOSE_TAG);
+      String key = rawKey;
+      String defaultValue = null;
+      if (rawKey.contains(SYSTEM_PROPERTY_DEFAULT_TAG)) {
+        String[] keyParts = rawKey.split(SYSTEM_PROPERTY_DEFAULT_TAG);
+        key = keyParts[0];
```

Comment:
I would like to do some checks here.  Specifically, require `key` to begin with an `metric` prefix (and maybe need to add `runtime` as well?).  Giving this class unfettered access to the system properties and environment variables can have security implications.  I think it would be safer to just eliminate access to any property or environment variable that doesn't start with a blessed prefix.

Reply:
Let's agree first if this is the right approach or not. I'm favored of getting rid of the system properties usage altogether.